tests/components/zoneminder/test_sensor.py

```diff
@@ -0,0 +1,167 @@
+        monitor2.is_recording = False
+        monitor2.is_available = False
+        monitor2.function = MonitorState.MODECT
+        monitor2.get_events.side_effect = _monitor2_get_events
+
+        zm_client: ZoneMinder = MagicMock(spec=ZoneMinder)
+        zm_client.get_zms_url.return_value = "http://host1/path_zms1"
+        zm_client.login.return_value = True
+        zm_client.get_monitors.return_value = [monitor1, monitor2]
+
+        zoneminder_mock.return_value = zm_client
+
+        config_entry = MockConfigEntry(
+            domain=const.DOMAIN,
+            unique_id="host1",
+            data={
+                CONF_HOST: "host1",
+                CONF_USERNAME: "username1",
+                CONF_PASSWORD: "password1",
+                CONF_PATH: "path1",
+                const.CONF_PATH_ZMS: "path_zms1",
+                CONF_SSL: False,
+                CONF_VERIFY_SSL: True,
+            },
+        )
+        config_entry.add_to_hass(hass)
+
+        hass_config = {
+            HASS_DOMAIN: {},
+            SENSOR_DOMAIN: [
+                {
+                    CONF_PLATFORM: const.DOMAIN,
+                    CONF_INCLUDE_ARCHIVED: True,
+                    CONF_MONITORED_CONDITIONS: ["all", "day"],
+                }
+            ],
+        }
+
+        await async_process_ha_core_config(hass, hass_config[HASS_DOMAIN])
+        await async_setup_component(hass, HASS_DOMAIN, hass_config)
+        await async_setup_component(hass, SENSOR_DOMAIN, hass_config)
+        await hass.async_block_till_done()
+        await async_setup_component(hass, const.DOMAIN, hass_config)
+        await hass.async_block_till_done()
+
+        await hass.services.async_call(
+            HASS_DOMAIN, "update_entity", {ATTR_ENTITY_ID: "sensor.monitor1_status"}
+        )
+        await hass.services.async_call(
+            HASS_DOMAIN, "update_entity", {ATTR_ENTITY_ID: "sensor.monitor1_events"}
+        )
+        await hass.services.async_call(
+            HASS_DOMAIN,
+            "update_entity",
+            {ATTR_ENTITY_ID: "sensor.monitor1_events_last_day"},
+        )
+        await hass.services.async_call(
+            HASS_DOMAIN, "update_entity", {ATTR_ENTITY_ID: "sensor.monitor2_status"}
+        )
+        await hass.services.async_call(
+            HASS_DOMAIN, "update_entity", {ATTR_ENTITY_ID: "sensor.monitor2_events"}
+        )
+        await hass.services.async_call(
+            HASS_DOMAIN,
+            "update_entity",
+            {ATTR_ENTITY_ID: "sensor.monitor2_events_last_day"},
+        )
+        await hass.async_block_till_done()
+        assert (
+            hass.states.get("sensor.monitor1_status").state
+            == MonitorState.MONITOR.value
+        )
+        assert hass.states.get("sensor.monitor1_events").state == "101"
+        assert hass.states.get("sensor.monitor1_events_last_day").state == "111"
+        assert hass.states.get("sensor.monitor2_status").state == "unavailable"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6184085)